3-UsingAPIs/Week2/homework/ex4-diceRace.js

```diff
@@ -1,32 +1,24 @@
 'use strict';
-/*------------------------------------------------------------------------------
-Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-UsingAPIs/Week2/README.md#exercise-4-dice-race
-
-1. Complete the function `rollDice()` by using `.map()` on the `dice` array 
-   to create an array of promises for use with `Promise.race()`.
-2. Refactor the function `main()` using async/await and try/catch.
-3. Once you got this working, you may observe that some dice continue rolling 
-   for some undetermined time after the promise returned by `Promise.race()` 
-   resolves. Do you know why? Add your answer as a comment to the bottom of the 
-   file.
-------------------------------------------------------------------------------*/
-// ! Do not remove this line
 const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
   const dice = [1, 2, 3, 4, 5];
-  // TODO complete this function; use Promise.race() and rollDie()
+  const promises = dice.map((die) => {
+    return rollDie(die);
+  });
```

Comment:
You are passing a function to `dice.map()` that takes a number and returns promise returned by `rollDie()`. But `rollDie()` itself also takes a number and returns the same promise. So no need for this intermediate (arrow) function:

```suggestion
  const promises = dice.map(rollDie);
```

You could even shorten it to:

```js
function rollDice() {
  const dice = [1, 2, 3, 4, 5];
  return Promise.race(dice.map(rollDie));
}
```